llvm/lib/Passes/PassBuilderPipelines.cpp

```diff
@@ -791,6 +792,7 @@ PassBuilder::buildFunctionSimplificationPipeline(OptimizationLevel Level,
 void PassBuilder::addRequiredLTOPreLinkPasses(ModulePassManager &MPM) {
   MPM.addPass(CanonicalizeAliasesPass());
   MPM.addPass(NameAnonGlobalPass());
+  MPM.addPass(AssignGUIDPass());
```

Comment:
this should be added as early as possible, so that (for example) PGO passes can use it. CtxProf definitely needs the GUID present.

Reply:
(after reading further below) it seems like the pass is used to ensure there are GUIDs. Should we call it "EnsureGUIDs" then?

Reply:
I think that just running it once as early as necessary is the way to go. I must admit I'm a bit lost amongst all the pass construction methods -- can you recommend the best place to put it?

Reply:
I believe `buildPerModuleDefaultPipeline` should be it.